apiserver/common/networkingcommon/types.go

```diff
@@ -132,6 +138,15 @@ type NetworkBacking interface {
 
 	// ReloadSpaces loads spaces from backing environ
 	ReloadSpaces(environ environs.BootstrapEnviron) error
+
+	// ReloadSpaces loads spaces from backing environ
+	SpaceByName(name string) (BackingSpace, error)
+
+	// AllEndpointBindings loads all endpointBindings.
+	AllEndpointBindings() ([]applicationcommon.ApplicationEndpoint, error)
```

Comment:
I had problems colocating the `Machine` interface in the shim because it is in package `spaces`, which imports `corenetwork`, which again would import `spaces`. 
This would be a cyclic dependency.

Is there another place where this type would make sense?

Reply:
This gets you built (including tests), but you have a test failure.
```diff
diff --git a/apiserver/common/networkingcommon/types.go b/apiserver/common/networkingcommon/types.go
index 1a1eab748a..a1fbe6d118 100644
--- a/apiserver/common/networkingcommon/types.go
+++ b/apiserver/common/networkingcommon/types.go
@@ -11,7 +11,6 @@ import (
diff --git a/apiserver/common/networkingcommon/types.go b/apiserver/common/networkingcommon/types.go
index 1a1eab748a..a1fbe6d118 100644
--- a/apiserver/common/networkingcommon/types.go
+++ b/apiserver/common/networkingcommon/types.go
@@ -11,7 +11,6 @@ import (
        "gopkg.in/juju/names.v3"

        "github.com/juju/juju/apiserver/common"
-       "github.com/juju/juju/apiserver/common/applicationcommon"
        "github.com/juju/juju/apiserver/params"
        "github.com/juju/juju/cloud"
        "github.com/juju/juju/core/life"
@@ -141,12 +140,6 @@ type NetworkBacking interface {

        // ReloadSpaces loads spaces from backing environ
        SpaceByName(name string) (BackingSpace, error)
-
-       // AllEndpointBindings loads all endpointBindings.
-       AllEndpointBindings() ([]applicationcommon.ApplicationEndpoint, error)
-
-       // AllMachines loads all machines
-       AllMachines() ([]Machine, error)
 }

 func BackingSubnetToParamsSubnet(subnet BackingSubnet) params.Subnet {
diff --git a/apiserver/facades/client/spaces/export_test.go b/apiserver/facades/client/spaces/export_test.go
index 37b114b8fc..fe51b16242 100644
--- a/apiserver/facades/client/spaces/export_test.go
+++ b/apiserver/facades/client/spaces/export_test.go
@@ -4,13 +4,12 @@
 package spaces

 import (
-       "github.com/juju/juju/apiserver/common/networkingcommon"
        "github.com/juju/juju/environs/context"
 )

 var NewAPIWithBacking = newAPIWithBacking

```

Reply:
@manadart ahhh thats what you mean I thought you meant moving the networkingcommon.Machine interface to the shim.

:smiley: 

Thanks for the patch!